src/main.ts

```diff
@@ -0,0 +1,89 @@
+import "./styles/style.css";
+import { getData } from "./util";
+
+type responseType = { key: string; ascii: string; audio: string }[];
+
+const URL =
+  "https://raw.githubusercontent.com/nikhilsnayak/static-data-files/main/drum-kit-data.json";
+const MAX_COUNT = 125;
+
+const keysContainer = document.querySelector<HTMLDivElement>(".keys")!;
+const audiosContainer = document.querySelector<HTMLDivElement>(".audios")!;
+const keyTemplate = <HTMLTemplateElement>(
+  document.getElementById("key-template")!
+);
+const audioTemplate = <HTMLTemplateElement>(
+  document.getElementById("audio-template")!
+);
+
+function renderElements(data: responseType): void {
+  data.forEach((item) => {
+    const key = keyTemplate.content.cloneNode(true) as HTMLElement;
+    const audio = audioTemplate.content.cloneNode(true) as HTMLElement;
+
+    key.querySelector(".key")!.setAttribute("data-key", item.ascii);
```

Comment:
prefer nil coalescing operator `?.`  here than the `!`.  ! will crash the program in case the expression which you are trying force extract the value via !

Reply:
In this case, I am confident that my markup includes an element with the class *key*. Therefore, I am assured of having an element, and using the ```!``` operator satisfies the TypeScript Compiler. In situations where there is no guarantee that an element with the class *key* will exist, I believe the ```?.``` operator would be the ideal choice, in my opinion. Please share your perspective on this.

Reply:
OK, one thing typescript is completely static time thing. So, it is very well possible that my html may be malformed. And this line might fail to return a proper node: ` key.querySelector(".key")` 
But yes, if you are 100% sure, then we can trip the compiler and force it to proceed.